DistroLauncher/ExitStatus.cpp

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "stdafx.h"
+#include <errno.h>
```

Comment:
is <cerrno> preferred for c++?

Reply:
It is not. I made that because when ifstream::open fails it doesn't tell exactly why so I'd need to look into the result of the system call. But after reflecting over your question I realized that I have std::filesystem in C++17 (even used it downwards ... 😅 ). I can avoid file-not-found error easily. That's what I'll do. Thanks for pointing that out.